src/pyhf/infer/calculators.py

```diff
@@ -188,6 +207,21 @@ def expected_value(self, nsigma):
         )
 
 
+@dataclass(frozen=True)
+class BestFitParameters:
```

Comment:
This could maybe be renamed to be something like `TestStatisticFitResults` to separate it from a single Fit

Reply:
To my ears `TestStatisticFitResults` sounds more like the theta-hat, theta-hat-hat tuple from a test statistic call.
Maybe `ConfidenceLevelFitResults` or `LimitFitResults` or `HypoTestFitResults`?